testing/test_pdb.py

```diff
@@ -434,3 +463,115 @@ def test_foo():
+
+    @pytest.mark.skipif(not SUPPORTS_BREAKPOINT_BUILTIN, reason="Requires breakpoint() builtin")
+    def test_sys_breakpointhook_configure_and_unconfigure(self, testdir):
+        """
+        Test that sys.breakpointhook is set to the custom Pdb class once configured, test that
+        hook is reset to system value once pytest has been unconfigured
+        """
+        config = testdir.parseconfig()
+
+        pytest_configure(config)
+        assert sys.breakpointhook == pytestPDB.set_trace
+
+        p1 = testdir.makepyfile("""
+        def test_nothing():
+            a = 0
+            assert a == 0
+        """)
+        result = testdir.runpytest_inprocess("", p1)
+        result.stdout.fnmatch_lines([
+            "* passed*",
+        ])
+        assert sys.breakpointhook != pytestPDB.set_trace
+
+    @pytest.mark.skipif(not SUPPORTS_BREAKPOINT_BUILTIN, reason="Requires breakpoint() builtin")
+    def test_sys_breakpointhook_configure_and_unconfigure_with_pdb_flag(self, testdir):
+        config = testdir.parseconfig()
+
+        pytest_configure(config)
+        assert sys.breakpointhook == pytestPDB.set_trace
+
+        p1 = testdir.makepyfile("""
+        def test_nothing():
+            a = 0
+            assert a == 0
+        """)
+        result = testdir.runpytest_inprocess("--pdb", p1)
+        result.stdout.fnmatch_lines([
+            "*1 passed*",
+        ])
+        assert sys.breakpointhook != pytestPDB.set_trace
+
+    @pytest.mark.skipif(not SUPPORTS_BREAKPOINT_BUILTIN, reason="Requires breakpoint() builtin")
+    def test_pdb_custom_cls(self, testdir, custom_debugger_hook):
+        p1 = testdir.makepyfile("""
+        def test_nothing():
+            breakpoint()
+        """)
+        result = testdir.runpytest_inprocess(
+            "--pdb", "--pdbcls=_pytest:_CustomDebugger", p1)
+        result.stdout.fnmatch_lines([
+            "*CustomDebugger*",
+            "*1 passed*",
+        ])
+        assert custom_debugger_hook == ["init", "set_trace"]
+
+    @pytest.mark.skipif(not SUPPORTS_BREAKPOINT_BUILTIN, reason="Requires breakpoint() builtin")
+    @pytest.mark.skipif(not _ENVIRON_PYTHONBREAKPOINT == '', reason="Requires breakpoint() default value")
+    def test_sys_breakpoint_interception(self, testdir):
+        p1 = testdir.makepyfile("""
+            def test_1():
+                breakpoint()
+        """)
+        child = testdir.spawn_pytest(str(p1))
+        child.expect("test_1")
+        child.expect("(Pdb)")
+        child.sendeof()
+        rest = child.read().decode("utf8")
+        assert "1 failed" in rest
+        assert "reading from stdin while output" not in rest
+        TestPDB.flush(child)
+
+    @pytest.mark.skipif(not SUPPORTS_BREAKPOINT_BUILTIN, reason="Requires breakpoint() builtin")
+    def test_pdb_not_altered(self, testdir):
+        """
+        Test that calling PDB set_trace() is not affected
+        when PYTHONBREAKPOINT=0
```

Comment:
Is this comment correct? I don't see `PYTHONBREAKPOINT` being set in the test..

Reply:
Removed in f1f4c8c1